### PR TITLE
Add cfn template for bastion

### PR DESCRIPTION
**Pull request by niroy**
This template launches a bastion host on customer vpc.
Instance has Mate AMI so you ssh and rdp both to it.

Attachment to transit gateway and tg route updates is currently automated, may have to be removed later so that learner manually does it .

We may automate adding the oc binary to the bastion host in future.

Everything has been tested so far and is working fine.

### CODE REVIEW

File: aws/cloudformation/rdp_bastion.yaml [hunk 1]
```diff
@@ -0,0 +1,492 @@
+---
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >
+  This template creates a bastion host based on Mate AMI on a new VPC(customer).
+  This bastion host will be used to connect to the ROSA cluster, run oc commands
+  and view the OpenShift console.
```
Comment (herve4m) [inline aws/cloudformation/rdp_bastion.yaml]: Missing space before the parenthesis:

```suggestion
  This template creates a bastion host based on Mate AMI on a new VPC
  (customer).
  This bastion host will be used to connect to the ROSA cluster, run oc commands
  and view the OpenShift console.
```

File: aws/cloudformation/rdp_bastion.yaml [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+    Type: AWS::EC2::NetworkAclEntry
+    Properties:
+      NetworkAclId:
+        Fn::GetAtt:
+          - rCustomerVpc
+          - DefaultNetworkAcl
+      RuleNumber: 60
+      Protocol: -1
+      RuleAction: deny
+      CidrBlock: 0.0.0.0/0
```
Comment (herve4m) [inline aws/cloudformation/rdp_bastion.yaml]: Those default network ACL were for the ROSA cluster. A customer VPC would not have those. Maybe remove all those ACLs and keep the defaults (allow everything), or deny everything (as it was the initial plan: https://drive.google.com/file/d/1PmRi_iKmiVR4pnHxJFJtEs-wEkURNRat/view?usp=sharing) and then add network ACL at the subnet level for just what you need.

**Review by herve4m — APPROVED** [review]

File: aws/cloudformation/rdp_bastion.yaml (hunk 2 above)
Comment (niroy) [reply aws/cloudformation/rdp_bastion.yaml]: Thanks, will change the nacl rules accordingly in the next commit.

File: aws/cloudformation/rdp_bastion.yaml [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+  This template creates a bastion host based on Mate AMI on a new VPC
+  (customer).
+  This bastion host will be used to connect to the ROSA cluster, run oc commands
+  and view the OpenShift console.
+
+Parameters:
+  pTagprefix:
+    Description: Use the course code to prepend resource tags
+    Type: String
+    Default: cs220
```
Comment (antmaro) [inline aws/cloudformation/rdp_bastion.yaml:13]:
```suggestion
    Default: cs220-cluster
```

Review the naming to fit the Herve's Cloudformation style

File: aws/cloudformation/rdp_bastion.yaml [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+  rCustomerVpc:
+    Type: AWS::EC2::VPC
+    Properties:
+      CidrBlock:
+        Ref: pCustomerVpcCidrBlock
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: !Join ["", [!Ref pTagprefix, "-customer-vpc"]]
```
Comment (antmaro) [inline aws/cloudformation/rdp_bastion.yaml]:
```suggestion
          Value: !Join ["", [!Ref pTagprefix, "-customer-vpc-bastion"]]
```

Review the changes from our conversation regarding bastion/vpn

**Review by antmaro — CHANGES_REQUESTED** [review]
As per our conversation about having two Customer VPCs VPN/Bastion, review my suggestions ;-)

File: aws/cloudformation/rdp_bastion.yaml (hunk 3 above)
Comment (niroy) [reply aws/cloudformation/rdp_bastion.yaml:13]: The customer vpc is outside the cs220-cluster (rosa) that's why I chose `cs220` and not `cs220-cluster`

Comment (niroy) [reply aws/cloudformation/rdp_bastion.yaml:13]: @herve4m and @antmaro can I keep the tag prefix as cs220? else I'll have to update it everywhere in the commands while retrieving the IDs of the resources.

Comment (niroy) [reply aws/cloudformation/rdp_bastion.yaml:13]: I'm keeping it as cs220 as it's easier to distinguish the resources of customer-vpc in AWS console

**Pull request merged**